core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -1479,6 +1508,48 @@ public Scope getAccessControlScope()
         }
     }
 
+    public static class SourceColumn
```

Comment:
You can add a convenience method to convert this class to `io.trino.spi.eventlistener.ColumnDetail`. So you could use it `io.trino.event.QueryMonitor#getQueryIOMetadata`